### ISSUE TITLE
Questions about re-implementation on other datasets

### ISSUE THREAD

**Issue by LinglanZhao** [RanPAC/RanPAC#1]
Hi!
Thank you for your transparent work! Could you provide the files of args/xxx_publish.csv for other datasets, such as imagenet-r, imagenet-a and vtab?

**Comment by drimpossible** [comment]
Seconded. Maybe I can provide more details of where I am stuck (only some ones are not reproducing currently, maybe I have some param wrong somewhere here?). Largely, the following files worked for me:

`imagenetr_publish.csv`
```
ID,dataset,shuffle,init_cls,increment,model_name,convnet_type,device,seed,batch_size,tuned_epoch,body_lr,head_lr,weight_decay,min_lr,use_RP,M
0,imagenetr,True,200,200,joint_linear,pretrained_vit_b16_224_in21k,0,1993,128,30,0.0,0.01,0.0005,0.0,False,-1
1,imagenetr,True,200,200,joint_full,pretrained_vit_b16_224_in21k,0,1993,128,30,0.0001,0.01,0.0005,0.0,False,-1
2,imagenetr,True,20,20,ncm,pretrained_vit_b16_224_in21k,0,1993,128,0,0.0,0.0,0.0005,0.0,False,-1
3,imagenetr,True,20,20,ncm,pretrained_vit_b16_224_in21k,0,1993,128,0,0.0,0.0,0.0005,0.0,True,0
4,imagenetr,True,20,20,ncm,pretrained_vit_b16_224_in21k,0,1993,128,0,0.0,0.0,0.0005,0.0,True,10000
5,imagenetr,True,20,20,adapter,pretrained_vit_b16_224_in21k_adapter,0,1993,48,20,0.01,0.01,0.0005,0.0,False,-1
6,imagenetr,True,20,20,adapter,pretrained_vit_b16_224_in21k_adapter,0,1993,48,20,0.01,0.01,0.0005,0.0,True,0
7,imagenetr,True,20,20,adapter,pretrained_vit_b16_224_in21k_adapter,0,1993,48,20,0.01,0.01,0.0005,0.0,True,10000
```
`imageneta_publish.csv`
```
ID,dataset,shuffle,init_cls,increment,model_name,convnet_type,device,seed,batch_size,tuned_epoch,body_lr,head_lr,weight_decay,min_lr,use_RP,M
0,imageneta,True,200,200,joint_linear,pretrained_vit_b16_224_in21k,0,1993,128,30,0.0,0.01,0.0005,0.0,False,-1
1,imageneta,True,200,200,joint_full,pretrained_vit_b16_224_in21k,0,1993,128,30,0.0001,0.01,0.0005,0.0,False,-1
2,imageneta,True,20,20,ncm,pretrained_vit_b16_224_in21k,0,1993,128,0,0.0,0.0,0.0005,0.0,False,-1
3,imageneta,True,20,20,ncm,pretrained_vit_b16_224_in21k,0,1993,128,0,0.0,0.0,0.0005,0.0,True,0
4,imageneta,True,20,20,ncm,pretrained_vit_b16_224_in21k,0,1993,128,0,0.0,0.0,0.0005,0.0,True,10000
5,imageneta,True,20,20,adapter,pretrained_vit_b16_224_in21k_adapter,0,1993,48,20,0.01,0.01,0.0005,0.0,False,-1
6,imageneta,True,20,20,adapter,pretrained_vit_b16_224_in21k_adapter,0,1993,48,20,0.01,0.01,0.0005,0.0,True,0
7,imageneta,True,20,20,adapter,pretrained_vit_b16_224_in21k_adapter,0,1993,48,20,0.01,0.01,0.0005,0.0,True,10000
```
`cub_publish.csv`
```
ID,dataset,shuffle,init_cls,increment,model_name,convnet_type,device,seed,batch_size,tuned_epoch,body_lr,head_lr,weight_decay,min_lr,use_RP,M
0,cub,True,200,200,joint_linear,pretrained_vit_b16_224_in21k,0,1993,128,30,0.0,0.01,0.0005,0.0,False,-1
1,cub,True,200,200,joint_full,pretrained_vit_b16_224_in21k,0,1993,128,30,0.0001,0.01,0.0005,0.0,False,-1
2,cub,True,20,20,ncm,pretrained_vit_b16_224_in21k,0,1993,128,0,0.0,0.0,0.0005,0.0,False,-1
3,cub,True,20,20,ncm,pretrained_vit_b16_224_in21k,0,1993,128,0,0.0,0.0,0.0005,0.0,True,0
4,cub,True,20,20,ncm,pretrained_vit_b16_224_in21k,0,1993,128,0,0.0,0.0,0.0005,0.0,True,10000
5,cub,True,20,20,adapter,pretrained_vit_b16_224_in21k_adapter,0,1993,48,20,0.01,0.01,0.0005,0.0,False,-1
6,cub,True,20,20,adapter,pretrained_vit_b16_224_in21k_adapter,0,1993,48,20,0.01,0.01,0.0005,0.0,True,0
7,cub,True,20,20,adapter,pretrained_vit_b16_224_in21k_adapter,0,1993,48,20,0.01,0.01,0.0005,0.0,True,10000
```
`omnibenchmark_publish.csv`
```
ID,dataset,shuffle,init_cls,increment,model_name,convnet_type,device,seed,batch_size,tuned_epoch,body_lr,head_lr,weight_decay,min_lr,use_RP,M
0,omnibenchmark,True,300,300,joint_linear,pretrained_vit_b16_224_in21k,0,1993,128,30,0.0,0.01,0.0005,0.0,False,-1
1,omnibenchmark,True,300,300,joint_full,pretrained_vit_b16_224_in21k,0,1993,128,30,0.0001,0.01,0.0005,0.0,False,-1
2,omnibenchmark,True,30,30,ncm,pretrained_vit_b16_224_in21k,0,1993,128,0,0.0,0.0,0.0005,0.0,False,-1
3,omnibenchmark,True,30,30,ncm,pretrained_vit_b16_224_in21k,0,1993,128,0,0.0,0.0,0.0005,0.0,True,0
4,omnibenchmark,True,30,30,ncm,pretrained_vit_b16_224_in21k,0,1993,128,0,0.0,0.0,0.0005,0.0,True,10000
5,omnibenchmark,True,30,30,adapter,pretrained_vit_b16_224_in21k_adapter,0,1993,48,20,0.01,0.01,0.0005,0.0,False,-1
6,omnibenchmark,True,30,30,adapter,pretrained_vit_b16_224_in21k_adapter,0,1993,48,20,0.01,0.01,0.0005,0.0,True,0
7,omnibenchmark,True,30,30,adapter,pretrained_vit_b16_224_in21k_adapter,0,1993,48,20,0.01,0.01,0.0005,0.0,True,10000
```
`vtab_publish.csv`
```
ID,dataset,shuffle,init_cls,increment,model_name,convnet_type,device,seed,batch_size,tuned_epoch,body_lr,head_lr,weight_decay,min_lr,use_RP,M
0,vtab,True,50,50,joint_linear,pretrained_vit_b16_224_in21k,0,1993,128,30,0.0,0.01,0.0005,0.0,False,-1
1,vtab,True,50,50,joint_full,pretrained_vit_b16_224_in21k,0,1993,128,30,0.0001,0.01,0.0005,0.0,False,-1
2,vtab,True,10,10,ncm,pretrained_vit_b16_224_in21k,0,1993,128,0,0.0,0.0,0.0005,0.0,False,-1
3,vtab,True,10,10,ncm,pretrained_vit_b16_224_in21k,0,1993,128,0,0.0,0.0,0.0005,0.0,True,0
4,vtab,True,10,10,ncm,pretrained_vit_b16_224_in21k,0,1993,128,0,0.0,0.0,0.0005,0.0,True,10000
5,vtab,True,10,10,adapter,pretrained_vit_b16_224_in21k_adapter,0,1993,48,20,0.01,0.01,0.0005,0.0,False,-1
6,vtab,True,10,10,adapter,pretrained_vit_b16_224_in21k_adapter,0,1993,48,20,0.01,0.01,0.0005,0.0,True,0
7,vtab,True,10,10,adapter,pretrained_vit_b16_224_in21k_adapter,0,1993,48,20,0.01,0.01,0.0005,0.0,True,10000
```
Accuracies Reproduced for RanPAC (`-i 7`)

ImageNetR- 77.03
ImageNetA- 52.83 (this didn't reproduce -- I tried a bunch of hparams but probably got something wrong?)
CUB- 90.13
VTAB- 92.72
Omnibenchmark- 79.73

Accuracies Reproduced for NCM (`-i 2`)

ImageNetR- 54.18 (this didn't reproduce -- I tried a bunch of hparams but probably got something wrong?)
ImageNetA- 48.55 
CUB- 86.75
VTAB- 85.88
Omnibenchmark- 73.3

Accuracies Reproduced for RANPAC without RP and without PETL (`-i 3`)

ImageNetR- 64.47 (this didn't reproduce -- I tried a bunch of hparams but probably got something wrong?)
ImageNetA- 50.46 (this didn't reproduce -- I tried a bunch of hparams but probably got something wrong?)
CUB- 85.87
VTAB- 92.0
Omnibenchmark- 73.06

Accuracies Reproduced for RANPAC without PETL (`-i 4`)

ImageNetR- 68.32 (this didn't reproduce -- I tried a bunch of hparams but probably got something wrong?)
ImageNetA- 51.76 (this didn't reproduce -- I tried a bunch of hparams but probably got something wrong?)
CUB- 89.94
VTAB- 92.88
Omnibenchmark- 78.25

**Comment by RanPAC** [comment]
Hi  @LinglanZhao  and @drimpossible - thank you both for your interest. The repository is now updated with args for 10 datasets. They also now have additional args for ResNet and CLIP results from the Appendix.

- All the args have been run using `python main_tables.py -d <dataset>`
- This create csv files in `paper_tables/` for all args in `args/<dataset>_publish/csv`
- I haven't checked all cases listed by @drimpossible but where I did check, the difference was in the choice of PETL method.
- Note that some experiments have greater variability from run to run than others. E.g. ssf in Phase 1
